include/Track.h

```diff
@@ -222,7 +228,7 @@ public slots:
 	BoolModel m_soloModel;
 	bool m_mutedBeforeSolo;
 
-	bool m_simpleSerializingMode;
+	bool m_presetMode;
```

Comment:
Can this variable be completely removed? It is only ever set to `true` right before loading and saving, and those routines just clear it at the end - so rather make it a parameter to these functions?

Reply:
Unfortunately that's not possible due to the design of the serialization. The variable would also need to go through `Track::saveSettings` and `Track::loadSettings` which in turn override the "generic" methods `SerializingObject::saveSettings` and `SerializingObject::loadSettings`.

The design more or less implies that each class knows how to serialize itself without using any outside information, like for example a preset mode, etc. Due to that it is necessary to bring each instance into a certain state using its "specialized" methods, like for example `setPresetMode`, so that it knows how to behave when being saved through that generic interface.

It feels a bit like OO design and polymorphism being used in the wrong context. But of course that's always easy to say in hindsight.

I have noted in the commit message of https://github.com/LMMS/lmms/pull/7445/commits/41da8043be215780097a4cc2cf556b275a48b715 that the existing implementation is rather brittle because after some exceptional situations an instance might accidentally stay in a certain mode.

Reply:
I have also experimented with adding an extra boolean to `Track::saveTrackSpecificSettings` and `Track::loadTrackSpecificSettings` but it did not really improve the code because then half of the code would use the member function while some other part would use the parameter. So I propose to keep it as is for now.

Reply:
If the idea is to let `Track::saveSettings(...)` call `Track::saveSettings(..., bool)` then how should the first one know what value to pass for the new boolean parameter to the second one?

My understanding is that the question was if it is possible to get rid of `setPresetMode` and `isPresetMode` altogether. This would mean that for example `InstrumentTrackWindow::saveSettingsBtnClicked` would need to indicate to the track in a different way that if should save as a preset, i.e. it could not call `m_track->setPresetMode()` anymore. If it does not call `setPresetMode` then the generic overridden method `saveSettings` does not know that it should call other methods related to preset mode in the first place.

Reply:
> If the idea is to let `Track::saveSettings(...)` call `Track::saveSettings(..., bool)` then how should the first one know what value to pass for the new boolean parameter to the second one?

`false`: The virtual function is (in my proposal) never called by the preset saving code. Same for loading.

Reply:
Moved the discussion about what else to do into the main thread.